### PR TITLE
DROOLS-6414 DMN Trisotech ext BoxedExpression extract or synth generic FEEL:list<T>

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6414

**referenced Pull Requests**: none

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-dmn/kie-dmn-trisotech/src/main/java/org/kie/dmn/trisotech/core/compiler/TrisotechDMNEvaluatorCompiler.java [hunk 1]
```diff
@@ -107,116 +90,96 @@ private DMNExpressionEvaluator compileConditional(DMNCompilerContext ctx, DMNMod
...
+            return new CompositeTypeImpl(orig.getNamespace(),
+                                         UUID.randomUUID().toString() + orig.getName(),
+                                         UUID.randomUUID().toString() + orig.getId(),
+                                         false, // synth T.
+                                         orig.getFields(),
+                                         null,
+                                         null);
+        }
+        return model.getTypeRegistry().unknown();
+    }
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-trisotech/src/main/java/org/kie/dmn/trisotech/core/compiler/TrisotechDMNEvaluatorCompiler.java:150]: this avoid introducing mutable changes in the DMN base type library (`kie-dmn-core`)
this avoid introducing third implementations of Types for the DMN base type library
this uses the generic `T` from a `FEEL:list<T>` when such baseType exists
while it creates a synthetic type when `T` is a Composite type AND at the same time directly a Collection (specific case being fixed with this PR too.)
@mswiderski ^

**Review by danielezonca — APPROVED** [review]

File: kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd [hunk 2]
```diff
@@ -14,6 +14,8 @@
 		</xsd:sequence>
 		<xsd:attribute name="id"
 			type="xsd:ID" use="optional" />
+		<xsd:attribute name="typeRef"
+			type="xsd:string" use="optional" />
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: @mswiderski why was this needed, considering:

https://github.com/kiegroup/drools/blob/3ff052f26fc66570ad58b3a2c02a87dc99ff9d7b/kie-dmn/kie-dmn-backend/src/test/resources/DMN13.xsd#L224-L229

is already defining the same attribute?

Comment (mswiderski) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: looks like a overlooked part as it seems to be duplicate

Comment (tarilabs) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: do you want to revise it on your end, since I would expect this PR will be checked locally from you anyway? :) 

I believe you should be able to send a PR against my branch `tarilabs:tarilabs-20210614-maciej`

Comment (mswiderski) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: yes, I will do that by the end of the day and comment back, thanks @tarilabs

Comment (mswiderski) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: @tarilabs so the change is actually needed as it extends the `tNamedExpression` and not the `tExpression`

Comment (tarilabs) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: this raises the question why `tNamedExpression` is not extending DMN's `tExpression` 🤔  can you check with @SimonRinguette please?

Comment (mswiderski) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: sure thing

Comment (tarilabs) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: I'm also fine to defer if any refactor possible in another separate PR, and have this one go through to fix your problem-at-hand, I just thought about raising this as eventually this will be proposed at the RTF I understand 🚀

**Review by mswiderski — APPROVED** [review]
works as expected

File: kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd (hunk 2 above)
Comment (mswiderski) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: we can do that, just approved as it does what was needed based on my tests so feel free to get it merged whenever you're ready

Comment (SimonRinguette) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: @tarilabs I did not want to have everything else that comes with it, we will need to "refactor" this for the submission to the spec for sure.

Comment (tarilabs) [reply kie-dmn/kie-dmn-trisotech/src/main/resources/org/kie/dmn/trisotech/validation/extension/TrisotechDMN13.xsd:18]: ack @SimonRinguette, thanks for the clarification

**Review by hellowdan — APPROVED** [review]
